### DISCUSSION TITLE
HotChocolate 13 type auto registration

### DISCUSSION THREAD

**Discussion by c0ngruens** [ChilliCream/hotchocolate#5845] [Q&A] [answered]
Hello!

I'm trying to implement type auto registration using HotChocolate [13.0.2](https://www.nuget.org/packages/HotChocolate/13.0.2) and [dotnet 7.0](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) as suggested [here](https://chillicream.com/blog/2023/02/08/new-in-hot-chocolate-13#type-auto-registration).

Tried configuring server using:

```csharp
builder.Services
    .AddGraphQLServer()
    .AddTypes();
```

Got error for `AddTypes`:

```
error CS0121: The call is ambiguous between the following methods or properties:
'SchemaRequestExecutorBuilderExtensions.AddTypes(IRequestExecutorBuilder, params Type[])' and
'SchemaRequestExecutorBuilderExtensions.AddTypes(IRequestExecutorBuilder, params INamedType[])'
```

Any assistance you can provide would be appreciated. Full example is [here](https://github.com/c0ngruens/HotChocolateAutoRegistrationTest).

Thanks

**Comment by c0ngruens** [comment, marked as answer]
How to implement auto registration can be found using template `dotnet new install HotChocolate.Templates`. I've updated my example [repo](https://github.com/c0ngruens/HotChocolateAutoRegistrationTest).

**Comment by FelixZY** [comment]
I had the same issue and was able to work it out:

1. You need to add the `HotChocolate.Types.Analyzers` dependency if you have not already done so.
2. You need to add `Properties/ModuleInfo.cs` with the line `[assembly: HotChocolate.Module("Types")]`
3. `AddTypes` is automatically generated at build time, meaning the error will go away once you `dotnet build`.

**Reply by FelixZY** [reply]
Also, it turns out you need to have at least one class for the auto registration to register. Otherwise, it will also fail with this error.

**Comment by promontis** [comment]
If it doesn't work, you probably have an out of date .NET SDK. Either update it, or update VS
